### PR TITLE
:goal_net: create error handler interface

### LINKED ISSUE #6: ⚠️ Error Handling
We should add an interface for Error management with `int getCode()` and `std::string getMessage()` to return the error message thrown, there will be 2 distinct interface, an IError for the Driver part and another for the Displayable part.

### PR DESCRIPTION
**Pull request by Antoinegtir**
create IError interface
create IGameError that inherit from IError
create IDriverError that inherit from Ierror

### CODE REVIEW

**Review by Marius-P1 — APPROVED** [review]
Good for me :ok:

**Review by axel-eck — APPROVED** [review]
Seems good to me

File: errors/IDriverError.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "../IDriver.hpp"
+#include "IError.hpp"
+
+class IDriverError: public IError {
+public:
+    virtual ~IDriverError() = default;
+    virtual int getCode() const = 0;
+    virtual std::shared_ptr<IDriver> getDriver() const = 0;
+    virtual const char *what() const noexcept = 0;
```
Comment (aimericdrk) [inline errors/IDriverError.hpp:17]: C style pointer

File: errors/IError.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#include <string>
+#include <memory>
+#include <exception>
+
+class IError: std::exception {
+public:
+    virtual ~IError() = default;
+    virtual int getCode() const = 0;
+    virtual const char *what() const noexcept = 0;
```
Comment (aimericdrk) [inline errors/IError.hpp:17]: C style pointer

File: errors/IGameError.hpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "../IGame.hpp"
+#include "IError.hpp"
+
+class IGameError: public IError {
+public:
+    virtual ~IGameError() = default;
+    virtual int getCode() const = 0;
+    virtual std::shared_ptr<IGame> getGame() const = 0;
+    virtual const char *what() const noexcept = 0;
```
Comment (aimericdrk) [inline errors/IGameError.hpp:17]: C style pointer

**Review by aimericdrk — APPROVED** [review]

File: errors/IDriverError.hpp (hunk 1 above)
Comment (axel-eck) [reply errors/IDriverError.hpp:17]: Look at what std::Exception wants.

File: errors/IError.hpp (hunk 2 above)
Comment (axel-eck) [reply errors/IError.hpp:17]: Look at what std::Exception wants.

File: errors/IGameError.hpp (hunk 3 above)
Comment (axel-eck) [reply errors/IGameError.hpp:17]: Look at what std::Exception wants.

**Review by Neiluge — APPROVED** [review]

**Pull request merged**